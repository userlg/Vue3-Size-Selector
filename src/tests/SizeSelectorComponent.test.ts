import { describe, it } from "vitest";

import { mount } from "@vue/test-utils";

import SizeSelectorVueComponent from "../components/SizeSelector.vue";


describe("sizeSelector Component", () => {

    test("mount app component", async () => {
        expect(SizeSelectorVueComponent).toBeTruthy();

        const wrapper = mount(SizeSelectorVueComponent, {
            props: {
                selectedSize: "M"
            }
        });

        expect(wrapper.text()).toContain("M");

        expect(wrapper.find('button').exists()).toBeTruthy();

        expect(wrapper.find('h3').exists()).toBeTruthy();

    });

    it("Render Correctly", async () => {

        const wrapper = mount(SizeSelectorVueComponent);

        expect(wrapper.html()).toMatchSnapshot();

    });

    it("Click in some button", async () => {

        const wrapper = mount(SizeSelectorVueComponent);

        await wrapper.find('button').trigger('click');

        expect(wrapper.emitted()).toHaveProperty('onChangedSize');
    });

});