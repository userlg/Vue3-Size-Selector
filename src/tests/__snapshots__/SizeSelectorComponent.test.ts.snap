// Vitest Snapshot v1

exports[`sizeSelector Component > Render Correctly 1`] = `
"<div class=\\"flex flex-row gap-1 sm:gap-2 items-center justify-center mx-auto\\"><button class=\\"bg-white w-1/2 sm:w-12 border border-black rounded p-2\\">
    <h3 class=\\"text-center text-xs sm:text-base\\">XS</h3>
  </button><button class=\\"bg-white w-1/2 sm:w-12 border border-black rounded p-2\\">
    <h3 class=\\"text-center text-xs sm:text-base\\">S</h3>
  </button><button class=\\"bg-white w-1/2 sm:w-12 border border-black rounded p-2\\">
    <h3 class=\\"text-center text-xs sm:text-base\\">M</h3>
  </button><button class=\\"bg-white w-1/2 sm:w-12 border border-black rounded p-2\\">
    <h3 class=\\"text-center text-xs sm:text-base\\">L</h3>
  </button><button class=\\"bg-white w-1/2 sm:w-12 border border-black rounded p-2\\">
    <h3 class=\\"text-center text-xs sm:text-base\\">XL</h3>
  </button><button class=\\"bg-white w-1/2 sm:w-12 border border-black rounded p-2\\">
    <h3 class=\\"text-center text-xs sm:text-base\\">XXL</h3>
  </button></div>"
`;
